xgcm/grid.py

```diff
@@ -843,12 +904,35 @@ def __init__(
                 axis_default_shifts = default_shifts[axis_name]
             else:
                 axis_default_shifts = {}
+
+            if isinstance(boundary, dict):
+                axis_boundary = boundary.get(axis_name, None)
```

Comment:
What does the user see if one of several axes is not included in the boundary dict? Should we warn here or further downstream?

Reply:
The default is `None` and the docstring says
```
* None:  Do not apply any boundary conditions. Raise an error if
              boundary conditions are required for the operation.
```

Seems fine to me?